src/main/scala/components/hazardmemstall.scala

```diff
@@ -0,0 +1,76 @@
+// This file contains the hazard detection unit
+
+package dinocpu
+
+import chisel3._
+
+/**
+ * The hazard detection unit, modified for memory stalling
+ *
+ * Input:  rs1, the first source register number
+ * Input:  rs2, the first source register number
+ * Input:  idex_memread, true if the instruction in the ID/EX register is going to read from memory
+ * Input:  idex_rd, the register number of the destination register for the instruction in the ID/EX register
+ * Input:  exmem_taken, if true, then we are using the nextpc in the EX/MEM register, *not* pc+4.
+ * Input:  imem_good, high if instruction memory is idle and ready.
```

Comment:
I thought "good" meant that the value on the output line was valid.

Reply:
It had to be changed, I will push a commit with a ready output representing the same signal. This will necessarily be a change in the stable interfaces